framework/core/src/Search/GambitManager.php

```diff
@@ -45,7 +45,12 @@ public function apply(SearchState $search, string $query): void
      */
     protected function explode(string $query): array
     {
-        return str_getcsv($query, ' ');
+        // Fixed Deprecated: str_getcsv(): Passing null to parameter #1 ($string) of type string is deprecated on php 8.x
+        if ($query !== null && is_string($query)) {
+            return str_getcsv($query, ' ');
+        } else {
+            return [];
+        }
```

Comment:
The `is_string` check is unnecessary because the method `explode` already type hints (enforces) `$query` as a `string`.

As you have targeted Flarum 2.x, the minimum php version is 8.1. As such we can simplify your code like this:

```suggestion
        return str_getcsv($query ?? '', ' ');
```

Although I think it's weird that `null` is being passed to `$query` as the type hint disallows that.